sopel/loader.py

```diff
@@ -232,24 +237,49 @@ def clean_module(module, config):
     callable, i.e. properties related to threading, docs, examples, rate
     limiting, commands, rules, and other features.
     """
-    callables = []
-    shutdowns = []
-    jobs = []
-    urls = []
+    callables: list[PluginCallable] = []
+    shutdowns: list = []
+    jobs: list[PluginJob] = []
+    urls: list[PluginCallable] = []
+    handler: PluginJob | PluginCallable
     for obj in vars(module).values():
         if callable(obj):
             is_sopel_callable = getattr(obj, '_sopel_callable', False) is True
             if getattr(obj, '__name__', None) == 'shutdown':
                 shutdowns.append(obj)
+                continue
             elif not is_sopel_callable:
                 continue
-            elif is_triggerable(obj):
-                clean_callable(obj, config)
-                callables.append(obj)
-            elif hasattr(obj, 'interval'):
-                clean_callable(obj, config)
-                jobs.append(obj)
-            elif is_url_callback(obj):
-                clean_callable(obj, config)
-                urls.append(obj)
+
+            if isinstance(obj, PluginJob):
+                obj.setup(config)
+
+                if obj.intervals:
+                    jobs.append(obj)
```

Comment:
I'm a little confused why things are done in this order, such that there are two `isinstance(var, PluginJob)` checks with slightly different behavior.

Would it be reasonable to massage objects that aren't yet `AbstractPluginObject` subclasses into the appropriate types first, _then_ run through the subclass checks to set them up?

Maybe you're doing it this way to better isolate the old "badly named" attributes from the new code. It's just a weird speed-bump reading through and seeing the same `isinstance()` check happen again below.